src/Button/index.js

```diff
@@ -1,79 +1,215 @@
 // @flow
-import React from "react";
+import React, { Component } from "react";
 import classNames from "classnames";
 import withStyles from "react-jss";
+import Color from "color";
+import Icon from "../Icon";
 
 type Props = {
-  element?: string,
   children: any,
   classes: Object,
   className: string,
-  color: "primary" | "secondary",
+  color: string,
+  element?: string,
+  icon: string,
   innerClassName?: string,
-  isRounded: true | false,
-  isOutlined: true | false,
+  isOutlined: boolean,
   theme: Object,
 };
 
-const Button = ({
-  children,
-  classes,
-  className,
-  color,
-  element: Element,
-  innerClassName,
-  isOutlined,
-  isRounded,
-  theme,
-  ...rest
-}: Props) => (
-  // $FlowFixMe -- it doesn't understand the element: Element
-  <Element className={classNames(classes.root, className)} {...rest}>
-    <div className={classNames(classes.children, innerClassName)}>
-      {children}
-    </div>
-  </Element>
-);
-
-Button.defaultProps = {
-  element: "button",
-  color: "primary",
-  isRounded: true,
-  isOutlined: false,
+type State = {
+  rippleActive: boolean,
 };
 
 function getChildColor(theme, props) {
   if (props.isDisabled) {
```

Comment:
Not sure if we have `isDisabled` anywhere else– but can we update this to use `disabled` and also make sure it gets added to the flow type definitions near the top?

Reply:
For sure, and it seems i missed some of the disabled states so I'll update those as well